server/homebrew.api.js

```diff
@@ -235,29 +235,26 @@ const updateBrew = async (req, res)=>{
 		brew.authors = _.uniq(_.concat(brew.authors, req.account.username));
 	}
 
-	// Fetch the brew from the database again (if it existed there to begin with), and assign the existing brew to it
-	brew = _.assign(await HomebrewModel.findOne({ _id: brew._id }), brew);
-
-	if(!brew.markModified) {
-		// If it wasn't in the database, create a new db brew
-		brew = new HomebrewModel(brew);
+	// define a function to catch our save errors
+	const saveError = (err)=>{
+		console.error(err);
+		res.status(err.status || 500).send(err.message || 'Unable to save brew to Homebrewery database');
+	};
+	let saved;
+	if(!brew._id) {
+		// if the brew does not have a stub id, create and save it, then write the new value back to the brew.
+		saved = await new HomebrewModel(brew).save().catch(saveError);
+		brew = saved?.toObject();
+	} else {
+		// if the brew does have a stub id, update it using the stub id as the key.
+		saved = await HomebrewModel.updateOne({ _id: brew._id }, brew).catch(saveError);
 	}
-
-	brew.markModified('authors');
-	brew.markModified('systems');
-
-	// Save the database brew
-	const saved = await brew.save()
-		.catch((err)=>{
-			console.error(err);
-			res.status(err.status || 500).send(err.message || 'Unable to save brew to Homebrewery database');
-		});
 	if(!saved) return;
 	// Call and wait for afterSave to complete
 	const after = await afterSave();
 	if(!after) return;
 
-	res.status(200).send(saved);
+	res.status(200).send(brew);
```

Comment:
Theoretically, `brew` and `saved` should be the same, but I think we still want to send `saved` because `editPage.jsx` uses that result for a couple things.

Reply:
So saved is either a brew, or an update result, using brew makes sure it's always the brew. updateOne doesn't return the brew (idk there might be a way to change that, lemme Google)

Reply:
Ah, so looking at `editPage.jsx`, at least one reason we need `saved` is if we are saving an unstubbed Google brew, `saved` will contain the newly-minted `editID` and `shareID`, which will be needed for subsequent saves.

We may need to use `findOneAndUpdate()` to get a return value , using the `options.returnDocument='after'` option to get the resulting document: https://mongoosejs.com/docs/api.html#model_Model-findOneAndUpdate

Reply:
So from what I can tell findOneAndUpdate returns the doc before it was modified https://mongoosejs.com/docs/tutorials/findoneandupdate.html#getting-started

Reply:
Yes but if you look down a little further you can set an option to return it after modifying.

Reply:
Oh cool! Okay yeah I'll do that. Sorry I'm on my phone right now 😅